### PR TITLE
Support for Protocol Buffers

### LINKED ISSUE #25: Protocol buffer interface
Suggestion by @mayhewsw : 

Add functionalities to convert `TextAnnotation` to/from protobuff: 
https://developers.google.com/protocol-buffers/

### PR DESCRIPTION
**Pull request by bhargav**
Adds support for serialization/deserialization to Protocol Buffers so that TextAnnotation can be used across platform/languages.

Only overhead is that structural changes to TextAnnotation must be accompanied by changes in JSON parsing and ProtoBuf parsing.

- [x] Add the ProtoBuf definition files.
- [x] Read/Write TextAnnotation in the new format.
- [x] Added basic unit test on top of JsonSerializationTest
- [x] Did run some benchmarks on loading/writing and ProtoBuf is about 4x faster than JSON and similar to the binary serialization.
- [x] Add instructions on how to compile the `TextAnnotation.proto` file. 

Not implementing:
- [x] ~Maven compile plugin?~ Plugin still needs the protobuf compiler installed locally. Added instructions to create the generated file.

Performance investigation at: https://wiki.illinois.edu/wiki/display/~mangipu2/2017/03/09/Protocol+Buffer+Investigations

Fixes #25

### CODE REVIEW

File: core-utilities/README.md [hunk 1]
```diff
@@ -401,6 +401,51 @@ to specify only non-default configuration options when you instantiate
...
+SerializationHelper.serializeTextAnnotationToProtobuf(ta, "text_annotation.buf", true);
+
+// Read file from disk and deserialize
+TextAnnotation ta = SerializationHelper.deserializeFromProtobuf("text_annotation.buf");
+```
+
+More usage information in the `SerializationHelper` class.
+
+###Generating Protocol Buffer Java Code
+Install the Protocol Buffer compiler locally. Run the following commands from the repository root:
```
Comment (danyaljj) [inline core-utilities/README.md]: Add a link to the "Buffer compiler"?

Comment (danyaljj) [inline core-utilities/README.md]: Add a little introduction here on, when you might need to "Generate the Protocol Buffer Java code"?: The answer can be like "if you change TextAnnotation, blah blah "

File: core-utilities/src/main/proto/TextAnnotation.proto [hunk 2]
```diff
@@ -0,0 +1,143 @@
+syntax = "proto2";
```
Comment (danyaljj) [inline core-utilities/src/main/proto/TextAnnotation.proto]: Do you want to add a few lines comments on top of this files saying, what is is, how it's generated, and when you will need to change it?

Comment (bhargav) [inline core-utilities/src/main/proto/TextAnnotation.proto]: Added comments.

File: core-utilities/README.md (hunk 1 above)
Comment (bhargav) [inline core-utilities/README.md]: Added a Note regarding changes to TextAnnotation.

**Comment by danyaljj** [comment]
@bhargav  Feel free to merge after being done with the comments.

**Pull request merged**